crates/router/src/connector/nuvei.rs

```diff
@@ -203,7 +203,14 @@ impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsR
         &self,
         req: &types::PaymentsCancelRouterData,
     ) -> CustomResult<Option<String>, errors::ConnectorError> {
-        let req_obj = nuvei::NuveiPaymentFlowRequest::try_from(req)?;
+        let req_obj = nuvei::NuveiPaymentFlowRequest::try_from(nuvei::NuveiPaymentRequestData {
```

Comment:
why we are creating `nuvei::NuveiPaymentRequestData` in the `get_request_body` of `ConnectorIntegration`.

It should refer from `req` only.

Reply:
previously it was called in the transformers as you said, because of [this comment](https://github.com/juspay/hyperswitch/pull/818#discussion_r1159568106) this was changed in `get_request_body`

Reply:
@Narayanbhat166 point is why we are doing two try from.

My point is, here we are doing transformation for `Void` flow with `PaymentsCancelRouterData`. In that case we should not transform  `NuveiPaymentFlowRequest` from `NuveiPaymentRequestData`.

It should be a transformation from `PaymentsCancelRouterData` to `NuveiPaymentFlowRequest`. This a template followed for transformation.

If transformation is common then we can have common function which can be called inside transformers.